src/main/java/frc/robot/subsystems/motorSubsystem/MotorSubsystem.java

```diff
@@ -0,0 +1,71 @@
+package frc.robot.subsystems.motorSubsystem;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.interfaces.ControllableMotor;
+import frc.robot.hardware.interfaces.IRequest;
+import frc.robot.hardware.interfaces.InputSignal;
+import frc.robot.subsystems.GBSubsystem;
+
+public class MotorSubsystem extends GBSubsystem {
+
+	private final ControllableMotor motor;
+	private final MotorCommandBuilder commandBuilder;
+	private IRequest<Rotation2d> velocityRequest;
+	private InputSignal<Rotation2d> velocitySignal;
+	private IRequest lastChangedRequest;
+
+	public MotorSubsystem(ControllableMotor motor, String logPath) {
+		super(logPath);
+		this.motor = motor;
+		this.commandBuilder = new MotorCommandBuilder(this);
+	}
+
+	public MotorSubsystem withVelocityControl(IRequest<Rotation2d> velocityRequest, InputSignal<Rotation2d> velocitySignal) {
+		this.velocityRequest = velocityRequest;
+		this.velocitySignal = velocitySignal;
+		return this;
+	}
+
+	public MotorCommandBuilder getCommandBuilder() {
+		return commandBuilder;
+	}
+
+	public boolean isAtVelocity(Rotation2d targetVelocity, Rotation2d tolerance) {
+		return MathUtil.isNear(targetVelocity.getRotations(), getVelocityRotation2dPerSecond().getRotations(), tolerance.getRotations());
+	}
+
+	public Rotation2d getVelocityRotation2dPerSecond() {
+		if (velocitySignal == null) {
```

Comment:
few problems: 
 - Null check without using optional -- other developers in this code won't know this var could be optional unless they would see this specific line. 
 - Throwing an exception -- it restarts the robot code. Imagine using this func in the pit, without proper checking, and the robot just keep restarting itself on the field. Or, a bug that occurs rarely, that we see first on the field. 

@YHerm, I was told that you instruct them to do so.

Reply:
critically having a class which can be just partially initialized is error prone. is there a reason this isnt passed in the constructor?